### ISSUE TITLE
Smoke test download can hang for 2 minutes

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#140892]
https://dev.azure.com/monacotools/Monaco/_build/results?buildId=150544&view=results

Log output:

```
[2022-01-18T07:35:51.438Z] Running desktop smoke tests against D:\a\1\VSCode-win32-x64\Code - Insiders.exe
[2022-01-18T07:35:51.486Z] Test data path: C:\Users\VSSADM~1\AppData\Local\Temp\vscsmoke
[2022-01-18T07:35:51.486Z] Preparing smoketest setup...
[2022-01-18T07:35:51.506Z] Starting operation 'versionReq...
[2022-01-18T07:35:51.506Z] Starting operation 'ensureStableCode...
[2022-01-18T07:35:51.723Z] Finished operation 'versionReq' successfully after 217ms
[2022-01-18T07:35:51.723Z] Starting operation 'versionReq.json()...
[2022-01-18T07:35:51.726Z] Finished operation 'versionReq.json()' successfully after 3ms
[2022-01-18T07:35:51.727Z] Found VS Code v1.64.0-insider, downloading previous VS Code version 1.63.2...
[2022-01-18T07:35:51.729Z] Starting operation 'download stable code...
[2022-01-18T07:35:51.920Z] download stable code progress: resolvedVersion
[2022-01-18T07:35:51.921Z] download stable code progress: resolvingCDNLocation
[2022-01-18T07:35:52.173Z] download stable code progress: downloading
[2022-01-18T07:37:51.517Z] Starting operation 'rimraf(testDataPath)...
[2022-01-18T07:37:51.523Z] Finished operation 'rimraf(testDataPath)' successfully after 6ms
```

So it looks like the download can just take 2 minutes for whatever reason. Maybe the progress could be improved further to log the amount of bytes transfered? And then maybe we have to accept that the smoke test needs a higher timeout.

Besides, could we just **cache** the stable code across multiple builds? Is that even possible?

**Comment by joaomoreno** [comment]
> Maybe the progress could be improved further to log the amount of bytes transfered?

Or rather, the error should report how far along the download was, when it timed out.

**Comment by Tyriar** [comment]
[Another one happened](https://dev.azure.com/monacotools/a6d41577-0fa3-498e-af22-257312ff0545/_build/results?buildId=150574), adding the trace here so I can easily find via search in the future:

```
  1) "before all" hook in "{root}"

  0 passing (2m)
  1 failing

  1) "before all" hook in "{root}":
     Error: Timeout of 120000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
      at listOnTimeout (internal/timers.js:557:17)
      at processTimers (internal/timers.js:500:7)
```

**Comment by connor4312** [comment]
Progressed will now be logged every 10s. Let me know if you need anything more from the vscode-test side.

**Comment by bpasero** [comment]
Thanks!

**Comment by bpasero** [comment]
Will tackle/reopen when I see this again and we have more data available.

**Comment by bpasero** [comment]
That was fast: https://dev.azure.com/monacotools/Monaco/_build/results?buildId=151114&view=logs&j=672276a2-8d3a-5fab-615d-090c51352f92&t=0699ae84-7245-5a45-5eee-80b086af2725&l=21

```
[2022-01-20T15:37:34.823Z] Running desktop smoke tests against D:\a\1\VSCode-win32-x64\Code - Insiders.exe
[2022-01-20T15:37:34.873Z] Test data path: C:\Users\VSSADM~1\AppData\Local\Temp\vscsmoke
[2022-01-20T15:37:34.873Z] Preparing smoketest setup...
[2022-01-20T15:37:34.897Z] Starting operation 'versionReq...
[2022-01-20T15:37:34.899Z] Starting operation 'ensureStableCode...
[2022-01-20T15:37:35.222Z] Finished operation 'versionReq' successfully after 325ms
[2022-01-20T15:37:35.223Z] Starting operation 'versionReq.json()...
[2022-01-20T15:37:35.225Z] Finished operation 'versionReq.json()' successfully after 2ms
[2022-01-20T15:37:35.226Z] Found VS Code v1.64.0-insider, downloading previous VS Code version 1.63.2...
[2022-01-20T15:37:35.229Z] Starting operation 'download stable code...
[2022-01-20T15:37:35.495Z] download stable code progress: resolvedVersion
[2022-01-20T15:37:35.495Z] download stable code progress: resolvingCDNLocation
[2022-01-20T15:37:35.809Z] download stable code progress: downloading (0/106120839)
[2022-01-20T15:37:45.811Z] download stable code progress: downloading (79708156/106120839)
[2022-01-20T15:42:34.922Z] Starting operation 'rimraf(testDataPath)...
[2022-01-20T15:42:34.926Z] Finished operation 'rimraf(testDataPath)' successfully after 4ms
```

@connor4312 any clues?

**Comment by bpasero** [comment]
Interestingly this always seems to be on Windows.

**Comment by connor4312** [comment]
Hm, so it looks like the data stream probably stopped receiving data. The logs are triggered every 10s, and no more logs means that there was no more `data` events coming from the download stream.

Either that, or the unzip process (which is run asynchronously to download) stalled for some reason. We did move to a new zip library in 1.6.0 which is faster but might be causing issues. In order to rule that out, I've added an option to use the old synchronous unzipping code path, and have toggled that on in this repo.
